examples/azure-service-mesh/security/cert-manager-letsencrypt/readme.md

```diff
@@ -0,0 +1,277 @@
+deployment.apps/reviews-v1 created
+deployment.apps/reviews-v2 created
+deployment.apps/reviews-v3 created
+service/productpage created
+serviceaccount/bookinfo-productpage created
+deployment.apps/productpage-v1 created
+```
+```shell
+kubectl get pods
+```
+Ensure each pod has 2/2 containers in the `Ready` state
+```console
+NAME                              READY   STATUS    RESTARTS   AGE
+details-v1-558b8b4b76-2llld       2/2     Running   0          2m41s
+productpage-v1-6987489c74-lpkgl   2/2     Running   0          2m40s
+ratings-v1-7dc98c7588-vzftc       2/2     Running   0          2m41s
+reviews-v1-7f99cc4496-gdxfn       2/2     Running   0          2m41s
+reviews-v2-7d79d5bd5d-8zzqd       2/2     Running   0          2m41s
+reviews-v3-7dbcdcbc56-m8dph       2/2     Running   0          2m41s
+```
+### 3. Configure ingress gateway and virtual service
+Before deploying the `virtualservice` and `gateway` resources, make sure to update the host name to match your own DNS name.
+```shell
+kubectl apply -f gateway.yaml
+kubectl apply -f virtualservice.yaml
+```
+
+> [!NOTE]  
+> In the gateway definition, `credentialName` must match the `secretName` in the `certificate` resource which will be created later in the example and selector must refer to the external ingress gateway by its label, in which the key of the label is `istio` and the value is `aks-istio-ingressgateway-external`.
+
+### Validate http request to productpage service
+
+Send an HTTP request to access the productpage service
+```shell
+curl -vs http://test.dev.azureservicemesh.io/productpage | grep -o "<title>.*</title>"
+```
+
+this should print `<title>Simple Bookstore App</title>`
+
+### 4. Create the shared configmap
+Create a ConfigMap with the name `istio-shared-configmap-<asm-revision>` in the `aks-istio-system` namespace to set `ingressService` and `ingressSelector`. For example, if your cluster is running asm-1-21 revision of mesh, then the ConfigMap needs to be named as istio-shared-configmap-asm-1-21. Mesh configuration has to be provided within the data section under mesh.
+
+```shell
+kubectl apply -f configmap.yaml
+```
+
+### 5. Install cert-manager
+```shell
+kubectl apply -f https://github.com/cert-manager/cert-manager/releases/download/v1.15.2/cert-manager.yaml
+```
+this installs cert-manager in its own namespace `cert-manager`
+
+#### Validate cert-manager pods
+```shell
+kubectl get po -n cert-manager
+```
+```console
+NAME                                     READY   STATUS    RESTARTS   AGE
+cert-manager-6fd987499c-xmf44            1/1     Running   0          72s
+cert-manager-cainjector-5b94bd6f-jw644   1/1     Running   0          72s
+cert-manager-webhook-575479ff47-d87pf    1/1     Running   0          72s
+```
+
+### 6. Setup cluster-issuer and Certificate resources
+Set your email address in `cluster-issuer.yaml` that you'd like to register with ACME server.
+```shell
+kubectl apply -f cluster-issuer.yaml
```

Comment:
Why the `IngressClass` is in this file ? Do you want to move it to a dedicated file to show the customer that this is also being created ?

Also, to avoid the customer editing your template files:

```suggestion
EMAIL=email
cat cluster-issuer.yaml | sed  -e "s/your-email-addr@domain.com/${EMAIL}/" | kubectl apply -f -
```